src/lib/blkalloc/varsize_blk_allocator.cpp

```diff
@@ -582,6 +582,8 @@ blk_count_t VarsizeBlkAllocator::alloc_blks_direct(blk_count_t nblks, blk_alloc_
     if (m_start_portion_num == INVALID_PORTION_NUM) { m_start_portion_num = m_rand_portion_num_generator(re); }
 
     auto portion_num = m_start_portion_num;
+    // save m_start_portion_num to local variable as m_start_portion_num can be changed by other threads.
+    auto start_portion_num = m_start_portion_num;
```

Comment:
I am not sure why we should have `m_start_portion_num`.
why can not we use a local var `start_portion_num = m_rand_portion_num_generator(re)` for each call of `alloc_blks_direct`. if i understand correctly, here we just only need a random start_portion for loop. coming to what is the start_portion, it does not matter, right?

Reply:
Yes technically it works.  These are two different initiative:

-   In current implementation (with m_start_portion_num shared) it tries to cache the last position we have successful allocation , as a result we tend to use up one portion first then comes to next.
- If we random the start portion on each allocation we tend to have a (statistically)  balance allocation across all regions.  

I dont have strong opinions so I didnt intent to change the behavior.